generator/integration_tests/golden/golden_client_test.cc

```diff
@@ -0,0 +1,702 @@
+  db = fut.get();
+  EXPECT_STATUS_OK(db);
+  EXPECT_EQ(expected_database, db->name());
+  EXPECT_EQ(::google::test::admin::database::v1::Database::CREATING,
+            db->state());
+}
+
+TEST(GoldenClientTest, GetDatabase) {
+  auto mock = std::make_shared<MockDatabaseAdminConnection>();
+  std::string expected_database(
+      "/projects/test-project/instances/test-instance/databases/test-db");
+  EXPECT_CALL(*mock, GetDatabase)
+      .Times(2)
+      .WillRepeatedly(
+          [expected_database](
+              ::google::test::admin::database::v1::GetDatabaseRequest const
+                  &request) {
+            EXPECT_EQ(expected_database, request.name());
+            ::google::test::admin::database::v1::Database response;
+            response.set_name(request.name());
+            return response;
+          });
+  DatabaseAdminClient client(std::move(mock));
+  auto response = client.GetDatabase(expected_database);
+  EXPECT_STATUS_OK(response);
+  EXPECT_EQ(response->name(), expected_database);
+  ::google::test::admin::database::v1::GetDatabaseRequest request;
+  request.set_name(expected_database);
+  response = client.GetDatabase(request);
+  EXPECT_STATUS_OK(response);
+  EXPECT_EQ(response->name(), expected_database);
+}
+
+TEST(GoldenClientTest, UpdateDatabase) {
+  auto mock = std::make_shared<MockDatabaseAdminConnection>();
+  std::string expected_database(
+      "/projects/test-project/instances/test-instance/databases/test-db");
+  EXPECT_CALL(*mock, UpdateDatabaseDdl)
+      .Times(2)
+      .WillRepeatedly([expected_database](
+                          ::google::test::admin::database::v1::
+                              UpdateDatabaseDdlRequest const &r) {
+        EXPECT_EQ(expected_database, r.database());
+        EXPECT_THAT(r.statements(), ElementsAre("-- test only: NOT SQL"));
+        ::google::test::admin::database::v1::UpdateDatabaseDdlMetadata metadata;
+        metadata.add_statements("-- test only: NOT SQL");
+        return make_ready_future(make_status_or(metadata));
+      });
+  DatabaseAdminClient client(std::move(mock));
+  auto fut =
+      client.UpdateDatabaseDdl(expected_database, {"-- test only: NOT SQL"});
+  ASSERT_EQ(std::future_status::ready, fut.wait_for(std::chrono::seconds(0)));
+  auto db = fut.get();
+  EXPECT_STATUS_OK(db);
+  EXPECT_THAT(db->statements(), ElementsAre("-- test only: NOT SQL"));
+  ::google::test::admin::database::v1::UpdateDatabaseDdlRequest request;
+  request.set_database(expected_database);
+  request.add_statements("-- test only: NOT SQL");
+  fut = client.UpdateDatabaseDdl(request);
+  ASSERT_EQ(std::future_status::ready, fut.wait_for(std::chrono::seconds(0)));
+  db = fut.get();
+  EXPECT_STATUS_OK(db);
+  EXPECT_THAT(db->statements(), ElementsAre("-- test only: NOT SQL"));
+}
+
+TEST(GoldenClientTest, DropDatabase) {
+  auto mock = std::make_shared<MockDatabaseAdminConnection>();
+  std::string expected_database(
```

Comment:
Total nit, but we'd normally write an initialization like this as `std::string foo = "bar"`

https://abseil.io/tips/88